index.php

```diff
@@ -175,6 +177,12 @@ public function __construct($baseDir) {
 		require_once $configFileName;
 		$this->configValues = $CONFIG;
 
+		if ($this->configValues['upgrade.disable-web'] ?? false) {
+			// updater disabled
+			$this->disabled = true;
```

Comment:
We could also throw an exception here like data directory. To check for web/cli we may pass PHP_SAPI  to the constructor.